### PR TITLE
add odoc-depgraph 1.0

### PR DESCRIPTION
**Pull request by zoggy**

### CODE REVIEW

File: packages/odoc-depgraph/odoc-depgraph.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+authors: ["Maxence Guesdon"]
+homepage: "https://github.com/zoggy/odoc_depgraph"
+license: "GNU Library General Public License version 2"
+doc: ["https://github.com/zoggy/odoc_depgraph"]
+tags: ["ocamldoc" "graph"]
+build: [
+  [make ]
+  [make "install"]
+]
+remove: [["rm" "-f" "/home/guesdon/.opam/4.02.1/lib/ocaml/ocamldoc/custom/odoc_depgraph.*"]]
```
Comment (avsm) [inline packages/odoc-depgraph/odoc-depgraph.1.0/opam]: This hardcodes your local path

Comment (zoggy) [inline packages/odoc-depgraph/odoc-depgraph.1.0/opam]: Oups :) This is fixed. Thanks.

**Comment by samoht** [comment]
wrong installation directory:
```
#=== ERROR while installing odoc-depgraph.1.0 =================================#
# opam-version 1.2.0
# os           linux
# command      make install
# path         /home/travis/.opam/system/build/odoc-depgraph.1.0
# compiler     system (4.02.1)
# exit-code    2
# env-file     /home/travis/.opam/system/build/odoc-depgraph.1.0/odoc-depgraph-16983-60459a.env
# stdout-file  /home/travis/.opam/system/build/odoc-depgraph.1.0/odoc-depgraph-16983-60459a.out
# stderr-file  /home/travis/.opam/system/build/odoc-depgraph.1.0/odoc-depgraph-16983-60459a.err
### stdout ###
# ocamlc -c -annot -I +ocamldoc `ocamlfind query -i-format dot` -safe-string odoc_depgraph.ml
# ocamlc -annot -I +ocamldoc `ocamlfind query -i-format dot` -safe-string -a -o odoc_depgraph.cma odot.cma odoc_depgraph.cmo
# ocamlopt -c -annot -I +ocamldoc `ocamlfind query -i-format dot` -safe-string odoc_depgraph.ml
# ocamlopt -annot -I +ocamldoc `ocamlfind query -i-format dot` -safe-string -shared -o odoc_depgraph.cmxs odot.cmxa odoc_depgraph.cmx
# mkdir -p `ocamldoc -customdir`
### stderr ###
# mkdir: cannot create directory `/usr/lib/ocaml/ocamldoc/custom': Permission denied
# make: *** [install] Error 1
```

**Comment by zoggy** [comment]
Hum, how am I supposed to call ocamldoc, then ?

**Comment by avsm** [comment]
Good question -- this might be the first OPAM ocamldoc plugin...

**Comment by samoht** [comment]
@zoggy do you have any suggestion? Is there an env variable we can set in opam (to point to something like `$prefix/ocamldoc/`?)

**Comment by zoggy** [comment]
Would it be better if I called `ocamlfind ocamldoc -customdir` instead of just `ocamldoc -customdir` ?

**Comment by zoggy** [comment]
Tried with using ocamlfind but this still points to /usr/[...] . But when a package tries to install in ocaml bin directory, it works, doesn't it ?

**Comment by samoht** [comment]
no, everything should be installed under `~/.opam/<switch>/{bin,lib}`

**Comment by zoggy** [comment]
But when I call ocamlfind or ocamldoc in ``make install``, both refer to ``/usr/lib/ocaml/`` (see the failed Travis build).

**Comment by zoggy** [comment]
Ok, looking closer at the error output, I understand that the system compiler is used. I thought that it would be a local one installed in `~/.opam/VERSION/`.

Installing in the `ocamldoc/custom` directory allows to use a custom generator with only the `-g` option of ocamldoc. If it is installed somewhere else, one must give the full path or use `-i` to specify an include directory (possibly using `ocamlfind query pkg`) in addition to `-i`.

So if the installation in `ocamldoc/custom` is a problem in opam, I will have to create a findlib package. Just tell me if ou think about a way to install in `ocamldoc/custom` or not.

**Comment by zoggy** [comment]
It seems ok now.

**Comment by samoht** [comment]
Thanks for fixing the issues!

**Pull request merged**